src/app/zap-templates/templates/chip/CHIPClusters-src.zapt

```diff
@@ -19,7 +19,6 @@ namespace Controller {
 {{#chip_server_cluster_commands}}
 CHIP_ERROR {{asCamelCased clusterName false}}Cluster::{{asCamelCased name false}}(Callback::Cancelable * onSuccessCallback, Callback::Cancelable * onFailureCallback{{#chip_server_cluster_command_arguments}}, {{chipType}} {{asCamelCased label}}{{/chip_server_cluster_command_arguments}})
 {
-#if CHIP_ENABLE_INTERACTION_MODEL
     VerifyOrReturnError(mDevice != nullptr, CHIP_ERROR_INCORRECT_STATE);
     (void) onSuccessCallback;
     (void) onFailureCallback;
```

Comment:
It seems like IM still does not support `onSuccessCallback` and `onErrorCallback`. This will break `iOS ChipTool`.

Reply:
For IM, we introduced interaction model delegate.

Reply:
Sure. But landing this PR that way will just break `iOS ChipTool`. I would prefer if you actually update the code to use the `Interaction Model Delegate` code path.

Also, it is not clear to me that `InteractionModelDelegate` (https://github.com/project-chip/connectedhomeip/blob/87142e056b579286b851b266fd855e93d103191d/src/app/InteractionModelDelegate.h#L39) carries the actual data that are passed to the mentioned callbacks ?

If not, can we keep the current codepaths for commands that have specific responses ?

Reply:
The IM will pass the status code, and command process status to the delegate.

You can refer to Python implementation for detail: https://github.com/project-chip/connectedhomeip/blob/87142e056b579286b851b266fd855e93d103191d/src/controller/python/chip/interaction_model/Delegate.cpp and https://github.com/project-chip/connectedhomeip/blob/87142e056b579286b851b266fd855e93d103191d/src/controller/python/chip/interaction_model/delegate.py

If the server response is another command, the command handler on the client side will be called.

Reply:
Added `DeviceControllerIMDelegate` so old code can still work, should be replaced by real IMDelegate impl later.